### PR TITLE
update Setonix config

### PR DESCRIPTION
**Pull request by BenWibking**
### Description
This updates the Setonix profile to use the latest module versions that are available. Pawsey updated the software stack in mid-June 2024 to Cray PE 23.09.

I've compiled and tested the ShockCloud problem on Setonix GPUs. It appears to work.

### Related issues
https://github.com/quokka-astro/quokka/issues/569

### Checklist
_Before this pull request can be reviewed, all of these tasks should be completed. Denote completed tasks with an `x` inside the square brackets `[ ]` in the Markdown source below:_
- [x] I have added a description (see above).
- [x] I have added a link to any related issues see (see above).
- [x] I have read the [Contributing Guide](https://github.com/quokka-astro/quokka/blob/development/CONTRIBUTING.md).
- [ ] I have added tests for any new physics that this PR adds to the code.
- [ ] I have tested this PR on my local computer and all tests pass.
- [ ] I have manually triggered the GPU tests with the magic comment `/azp run`.
- [x] I have requested a reviewer for this PR.

### CODE REVIEW

File: src/ShockCloud/cloud.cpp
Comment (markkrumholz) [inline src/ShockCloud/cloud.cpp:1]: Did you intend to include this change in this PR? It seems unrelated.

**Review by markkrumholz — COMMENTED** [review]
See one minor comment.

File: src/ShockCloud/cloud.cpp
Comment (BenWibking) [reply src/ShockCloud/cloud.cpp:1]: It's necessary to run it on AMD. Otherwise, it fails at runtime due to a compiler bug.

Comment (markkrumholz) [reply src/ShockCloud/cloud.cpp:1]: OK, understood.

**Review by markkrumholz — APPROVED** [review]
